fix: direnv individual message styling (#6153)

0x57e11a opened the pull request:
#### Description
permits the `direnv` module's `*_msg` segments to use the formatter:
- switches from `map` to `map_meta`
- introduces a warning if the `rc_path` isn't UTF-8 (reasonably this shouldn't be an issue?)

#### Motivation and Context
as detailed in #6153, it's nice to have colors for quick glancing, and not having the ability to format segments is an inconsistency with other modules

closes #6153

#### Screenshots (if appropriate):
using the following config:
```toml
[direnv]
disabled = false

allowed_msg = "[allowed](bold green)"
not_allowed_msg = "[not allowed](bold yellow)"
denied_msg = "[denied](bold red)"

loaded_msg = "[loaded](bold green)"
unloaded_msg = "[unloaded](bold red)"

format = "$allowed $loaded"
```
before:
![image](https://github.com/user-attachments/assets/7649263a-62cd-48de-9780-d0312844a8c5)
after:
![image](https://github.com/user-attachments/assets/89cc3f0c-d0c7-4cf4-bd30-23980ed9b92d)

#### How Has This Been Tested?
- the `msg_formatting` test in the `direnv` module passes on linux
- the above screenshots/config 

testing should be trivial on other platforms, but these changes shouldn't be platform-dependent
- [ ] I have tested using **MacOS**
- [x] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:
- [x] I have updated the documentation accordingly.
  likely not necessary, as none of the other modules mention that their variables can also be formatted strings
- [x] I have updated the tests accordingly.

0x57e11a committed:
fix: direnv individual message styling (#6153)

0x57e11a committed:
test: add direnv msg_formatting unit test

0x57e11a commented:
if really necessary, it shouldn't be too difficult to write a test to [cover the non-utf8 case](https://github.com/starship/starship/pull/6389/files#diff-4294d1cb69edbfa3b451b59515264b4506b03dac9dd15b629d9013d9746ec36aR46-R50)

davidkna commented:
I would prefer merging #6338 first to avoid this causing issues for conditionals. 

Also, please just keep the `rc_path`  behaviour as-is and refrain from moving the other variables to `map_meta` and keepthe rest in `map`.

0x57e11a committed:
fix(direnv): move `symbol` & `rc_path` into `map`

resolves: https://github.com/starship/starship/pull/6389#issuecomment-2480559116

0x57e11a commented:
> I would prefer merging #6338 first to avoid this causing issues for conditionals.

sounds good!

> Also, please just keep the `rc_path` behaviour as-is and refrain from moving the other variables to `map_meta` and keep the rest in `map`.

moved!
